src/infrastructure/validation/testing/TestHarness.java

```diff
@@ -0,0 +1,242 @@
+      Object test = testClass.getDeclaredConstructor().newInstance();
+      if(priority == (test.getPriority())){
+         boolean result = test.run(); 
+         
+         if(result == false){
+          failedNumberOfTests++;
+          if(failedNumberOfTests == 1){
+            logger.log(ModuleID.TEST, LogLevel.INFO, "Failed Tests...");
+          }
+          //log the result of failed test cases
+          logger.log(ModuleID.TEST, LogLevel.INFO, Integer.toString(failedNumberOfTests)+". "+testName+": "+test.getError());
+         }
+         else{
+            successfulNumberOfTests++;
+         }
+        
+      }  
+    } 
+      
+    //result logging
+    logger.log(ModuleID.TEST, LogLevel.INFO, "\nOverall Result:");
+    logger.log(ModuleID.TEST, LogLevel.INFO, "Total Number of Tests: "+Integer.toString(totalNumberOfTests));
+    logger.log(ModuleID.TEST, LogLevel.INFO, "Total Number of Successful Tests: "+Integer.toString(successfulNumberOfTests));
+    logger.log(ModuleID.TEST, LogLevel.INFO, "Total Number of Failed Tests: "+Integer.toString(failedNumberOfTests));
+  }
+
+
+ /**
+  * Run all test cases of all modules one by one 
+  */
+  public void runAll(){
+    int totalNumberOfTests = 0;
+    int successfulNumberOfTests = 0;
+    int failedNumberOfTests = 0;
+    
+    ILogger logger = LoggerFactory.getLoggerInstance();
+     
+    //call the helper method to get list of all test cases 
+    ArrayList<File> allTests = getAllTests();
+    totalNumberOfTests = allTests.size();
+    
+    for (File testFile : allTests){
+      String testName = testFile.getName();
+      String[] arrOfStr = testName.split(".", 2); 
+      String testClassName = arrOfStr[0];
+      Class<?> testClass = Class.forName("testClassName");
+      Object test = testClass.getDeclaredConstructor().newInstance();
+      boolean result = test.run();  
+       
+      if(result == false){
+        failedNumberOfTests++;
+        if(failedNumberOfTests == 1){
+          logger.log(ModuleID.TEST, LogLevel.INFO, "Failed Tests...");
+        }
+        //log the result of failed test cases
+        logger.log(ModuleID.TEST, LogLevel.INFO, Integer.toString(failedNumberOfTests)+". "+testName+": "+test.getError());
+       }
+       else{
+        successfulNumberOfTests++;
+       }
+    }
+    
+    //result logging 
+    logger.log(ModuleID.TEST, LogLevel.INFO, "\nOverall Result:");
+    logger.log(ModuleID.TEST, LogLevel.INFO, "Total Number of Tests: "+Integer.toString(totalNumberOfTests));
+    logger.log(ModuleID.TEST, LogLevel.INFO, "Total Number of Successful Tests: "+Integer.toString(successfulNumberOfTests));
+    logger.log(ModuleID.TEST, LogLevel.INFO, "Total Number of Failed Tests: "+Integer.toString(failedNumberOfTests));
+  } 
+	 
+	  
+ /**
+  * Boolean method to give result of the Test Case Class
+  * @param testName The absolute path to file name of test case class including .java extension as a string
+  */
```

Comment:
You are asking for absolute path and then split the given string like just the test name is given. The ideal behaviour is that only test name is passed and test harness figures out where the file is. The getAllTests will come in handy I guess

Reply:
We will have problem if two test classes have same name.